### PR TITLE
Initialise slasher service using New in slasher test to avoid panics

### LINKED ISSUE #11045: Panic when running slasher benchmark
# 🐞 Bug Report

### Description

When running the slasher benchmark I get a `panic: assignment to entry in nil map`

The line mentioned in the stack accesses the map `latestEpochWrittenForValidator` in the slasher service. As the tests and the benchmark do not create the service with `New` they do not initialise all fields.

### Has this worked before in a previous version?

I guess. I don't know

## 🔬 Minimal Reproduction

```
cd beacon-chain/slasher
go test -bench=.
```

## 🔥 Error

<pre><code>
panic: assignment to entry in nil map

goroutine 195 [running]:
github.com/prysmaticlabs/prysm/beacon-chain/slasher.(*Service).checkSlashableAttestations(0xc0000e0680, {0x1a3a768, 0xc0000400c0}, 0x1, {0xc00030a000, 0x1, 0x1})
        /home/leo/src/prysm/beacon-chain/slasher/detect_attestations.go:48 +0xdc5
github.com/prysmaticlabs/prysm/beacon-chain/slasher.runAttestationsBenchmark(0xc00035c480, 0xc0000e0680, 0x1, 0x1)
        /home/leo/src/prysm/beacon-chain/slasher/detect_attestations_test.go:890 +0x205
github.com/prysmaticlabs/prysm/beacon-chain/slasher.BenchmarkCheckSlashableAttestations.func1(0xc00035c480?)
        /home/leo/src/prysm/beacon-chain/slasher/detect_attestations_test.go:826 +0x3e
testing.(*B).runN(0xc00035c480, 0x1)
        /usr/lib/go-1.18/src/testing/benchmark.go:193 +0x102
testing.(*B).run1.func1()
        /usr/lib/go-1.18/src/testing/benchmark.go:233 +0x59
created by testing.(*B).run1
        /usr/lib/go-1.18/src/testing/benchmark.go:226 +0x9c
exit status 2
FAIL    github.com/prysmaticlabs/prysm/beacon-chain/slasher     0.060s
</code></pre>

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
   Ubuntu 22.04 on AMD64
  </code>
</pre>

**What version of Prysm are you running? (Which release)**

<pre>
  <code>
    develop branch
  </code>
</pre>

**Anything else relevant (validator index / public key)?**

No

### PR DESCRIPTION
**Pull request by leolara**
**What type of PR is this?**

> Bug fix

**What does this PR do? Why is it needed?**

Fixes the initialisation of the slasher service in its tests, so it does not panic when a field is nil when it should be initialised.

**Which issues(s) does this PR fix?**

Fixes #11045 - Panic when running slasher benchmark

**Other notes for review**

If you do not want to initialise the service in the tests using `New` let me know and I will fix it adding the initialisation of that field explicitly in the test that needs it.

### CODE REVIEW

**Review by rkapka — DISMISSED** [review]

File: beacon-chain/slasher/detect_attestations_test.go [hunk 1]
```diff
@@ -810,15 +801,11 @@ func BenchmarkCheckSlashableAttestations(b *testing.B) {
...
-	s := &Service{
-		params: DefaultParams(),
-		serviceCfg: &ServiceConfig{
-			Database:         slasherDB,
-			StateNotifier:    &mock.MockStateNotifier{},
-			HeadStateFetcher: mockChain,
-		},
-		attsQueue: newAttestationsQueue(),
-	}
+	s, _ := New(context.Background(), &ServiceConfig{
```
Comment (prestonvanloon) [inline beacon-chain/slasher/detect_attestations_test.go:804]:
```
beacon-chain/slasher/detect_attestations_test.go:804:5: Unhandled error for function call  (errcheck)
```

This error must be handled

Comment (leolara) [reply beacon-chain/slasher/detect_attestations_test.go:804]: ok, will fix today. It was a mistake the other are handled

Comment (leolara) [reply beacon-chain/slasher/detect_attestations_test.go]: done

**Review by prestonvanloon — APPROVED** [review]

**Comment by prestonvanloon** [comment]
Thank you!

**Pull request merged**